src/language-html/utils.js

```diff
@@ -14,6 +14,27 @@ const htmlElementAttributes = require("html-element-attributes");
 const HTML_TAGS = arrayToMap(htmlTagNames);
 const HTML_ELEMENT_ATTRIBUTES = mapObject(htmlElementAttributes, arrayToMap);
 
+// https://infra.spec.whatwg.org/#ascii-whitespace
+const HTML_WHITESPACE = new Set(["\t", "\n", "\f", "\r", " "]);
+const htmlTrimStart = (string) => string.replace(/^[\t\n\f\r ]+/, "");
+const htmlTrimEnd = (string) => string.replace(/[\t\n\f\r ]+$/, "");
+const htmlTrim = (string) => htmlTrimStart(htmlTrimEnd(string));
+const htmlTrimLineByLine = (string) =>
+  string.replace(/^[\t\f\r ]*?\n|\n[\t\f\r ]*?$/g, "");
```

Comment:
@thorn0 

This function seem not "trimLineByLine",

old code `value.replace(/^\s*?\n|\n\s*?$/g, "")`,

more like `removeEmptyLine`?

Reply:
No it's not, I really don't understand...

Reply:
```
`

\u0020

  a \u0020
     
b \u0020
\u0020
c
\u0020
`.replace(/^\s*?\n|\n\s*?$/g, "").replace(/ /g, '_')
"
_

__a__
_____
b__
___
c"
```

Reply:
`\u0020\u0020\u0020\u0020

\u0020\u0020

  a \u0020
     
b \u0020
\u0020
c
\u0020\u0020\u0020
`.replace(/^\s*?\n|\n\s*?$/g, "").replace(/ /g, '_')
"
__

__a__
_____
b__
___
c"

`trimFirstLineAndTrailingWhitespace`?

Reply:
Updated to `htmlTrimFirstLine` and `htmlTrimEnd` seems work https://github.com/prettier/prettier/pull/8137/commits/9a9d36f3ed9e5cb994e09146efcad3704943efc5

Reply:
Doesn't look like an equivalent replacement.

Reply:
Let  me  ask the  author :)

@ikatyang , you first wrote this  in

https://github.com/prettier/prettier/commit/b52be6bdcad89c1af1239cf10aed4d235a5c4c20#diff-3e8cd8ca0d07925abf364c9d8a413220R143


Do  you  still  remember what is suppose to do ?

Reply:
Okay, figured it out. This code removes leading and trailing blank lines (lines consisting only of whitespace), but preserves the leading whitespace of the first non-blank line and the trailing whitespace of the last non-blank line. The leading whitespace is needed to correctly determine the indentation of scripts (related to #8147, that's why vbscript is indented). And the trailing whitespace doesn't really seem to be of any use, so your replacement looks good. I'd only suggest renaming `htmlTrimFirstLine` to `htmlTrimLeadingBlankLines`.

I'm still not sure it's okay to indent those scripts, but so far nobody complained, and that's a separate issue anyway.